subsys/usb/usb_c/usbc_pe_src_states.c

```diff
@@ -0,0 +1,702 @@
+			switch (header.message_type) {
+			case PD_DATA_REQUEST:
+				pe_set_state(dev, PE_SRC_NEGOTIATE_CAPABILITY);
+				break;
+			default:
+				pe_set_state(dev, PE_SEND_NOT_SUPPORTED);
+			}
+		}
+		/*
+		 * Control Message Requests
+		 */
+		else {
+			switch (header.message_type) {
+			case PD_CTRL_GOOD_CRC:
+				/* Do nothing */
+				break;
+			case PD_CTRL_NOT_SUPPORTED:
+				/* Do nothing */
+				break;
+			case PD_CTRL_PING:
+				/* Do nothing */
+				break;
+			case PD_CTRL_GET_SOURCE_CAP:
+				pe_set_state(dev, PE_SRC_SEND_CAPABILITIES);
+				break;
+				break;
+			case PD_CTRL_DR_SWAP:
+				pe_set_state(dev, PE_DRS_EVALUATE_SWAP);
+				break;
+			/*
+			 * USB PD 3.0 6.8.1:
+			 * Receiving an unexpected message shall be responded
+			 * to with a soft reset message.
+			 */
+			case PD_CTRL_ACCEPT:
+			case PD_CTRL_REJECT:
+			case PD_CTRL_WAIT:
+			case PD_CTRL_PS_RDY:
+				pe_send_soft_reset(dev, prl_rx->emsg.type);
+				break;
+				/*
+				 * Receiving an unknown or unsupported message
+				 * shall be responded to with a not supported
+				 * message.
+				 */
+			default:
+				pe_set_state(dev, PE_SEND_NOT_SUPPORTED);
+				break;
+			}
+		}
+	} else {
+		/* Handle Source DPManager Requests */
+		source_dpm_requests(dev);
+	}
+}
+
+/**
+ * @brief 8.3.3.2.11 PE_SRC_Transition_to_default State
+ */
+void pe_src_transition_to_default_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * On entry to the PE_SRC_Transition_to_default state the
+	 * Policy Engine Shall:
+	 *	1: indicate to the Device Policy Manager that the power
+	 *	   supply Shall Hard Reset
+	 *	2: request a reset of the local hardware
+	 *	3: request the Device Policy Manager to set the Port
+	 *	   Data Role to DFP and turn off VCONN.
+	 *
+	 * NOTE: 1, 2 and VCONN off are done by Device Policy Manager when
+	 *	 it receives the HARD_RESET_RECEIVED notification.
+	 */
+	policy_notify(dev, HARD_RESET_RECEIVED);
+	pe->data_role = TC_ROLE_DFP;
+	policy_notify(dev, DATA_ROLE_IS_DFP);
+}
+
+void pe_src_transition_to_default_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * The Policy Engine Shall transition to the PE_SRC_Startup
+	 * state when:
+	 *	1: The Device Policy Manager indicates that the power
+	 *	   supply has reached the default level.
+	 */
+	if (policy_check(dev, CHECK_SRC_PS_AT_DEFAULT_LEVEL)) {
+		pe_set_state(dev, PE_SRC_STARTUP);
+	}
+}
+
+void pe_src_transition_to_default_exit(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	/*
+	 * On exit from the PE_SRC_Transition_to_default state the
+	 * Policy Engine Shall:
+	 *	1: request the Device Policy Manager to turn on VCONN
+	 *	2: inform the Protocol Layer that the Hard Reset is complete.
+	 *
+	 * NOTE: The Device Policy Manager turns on VCONN when it send the
```

Comment:
"The Device Policy Manager turns on VCONN when it notifies the PE that Power Supply is at the default level"?

Reply:
fixed